lib/Serialization/Deserialization.cpp

```diff
@@ -1770,12 +1770,34 @@ Identifier ModuleFile::getIdentifier(IdentifierID IID) {
 
   assert(!IdentifierData.empty() && "no identifier data in module");
 
-  StringRef rawStrPtr = IdentifierData.substr(identRecord.Offset);
-  size_t terminatorOffset = rawStrPtr.find('\0');
-  assert(terminatorOffset != StringRef::npos &&
-         "unterminated identifier string data");
+  auto flag = IdentifierData.data()[identRecord.Offset];
+  if (flag == '\0') {
+    // Normal DeclName
+    StringRef rawStrPtr = IdentifierData.substr(identRecord.Offset + 1);
+    size_t terminatorOffset = rawStrPtr.find('\0');
+    assert(terminatorOffset != StringRef::npos &&
+           "unterminated identifier string data");
+
+    return getContext().getIdentifier(rawStrPtr.slice(0, terminatorOffset));
+  } else if (flag == '\1') {
+    // Subscript
+    return DeclName::createSubscript();
+  } else {
+    // Serialized prior to special DeclNames. Just read the normal identifier
```

Comment:
We're treating the minor version as a breaking version for now, please continue doing that.